sentinel/src/schedule/replications.js

```diff
@@ -37,11 +43,102 @@ function getSchedule(config) {
   }
 }
 
+const purgeDocs = (sourceDb, docs) => {
+  if (!docs || !docs.length) {
+    return Promise.resolve();
+  }
+
+  const docsToPurge = {};
+  docs.forEach(doc => docsToPurge[doc._id] = [doc._rev]);
+
+  return sourceDb
+    .info()
+    .then(info => {
+      const opts = {
+        uri: `${db.serverUrl}/${info.db_name}/_purge`,
+        json: true,
+        body: docsToPurge,
+      };
+
+      return rpn.post(opts);
+    });
+};
+
+const getHasRunOneTimePurge = (sourceDb) => {
+  return sourceDb
+    .get(ONE_TIME_PURGE_LOCAL_DOC_ID)
+    .then(() => true)
+    .catch(err => {
+      if (err.status === 404) {
+        return false;
+      }
+      throw err;
+    });
+};
+
+const oneTimePurgeRan = (sourceDb) => {
+  return sourceDb.put({ _id: ONE_TIME_PURGE_LOCAL_DOC_ID });
+};
+
+const oneTimePurge = (sourceDb, targetDb) => {
+  return getHasRunOneTimePurge(sourceDb)
+    .then(hasRunOneTimePurge => {
+      if (hasRunOneTimePurge) {
+        return;
+      }
+
+      return batchedPurge(sourceDb, targetDb).then(() => oneTimePurgeRan(sourceDb));
+    });
+};
+
+const batchedPurge = (sourceDb, targetDb, lastSeq = 0) => {
+  const opts = {
+    since: lastSeq,
+    limit: BATCH_SIZE,
```

Comment:
Should we do two BATCH_SIZES? One for getting changes (likely larger) and then mini-batch the purges?

Reply:
I think it's ok. It'd be a little more efficient to get a larger batch, but the code would be more complex.